db-config.php

```diff
@@ -284,37 +284,48 @@ function my_db_callback($query, $wpdb) {
  * @param string $password Database password.
  */
 /*
-function add_db_server($dataset, $part, $dc, $read, $write, $host, $lhost, $name, $user, $password, $timeout = 0.2 ) {
+function add_db_server( $dataset, $part, $dc, $read, $write, $host, $lhost, $name, $user, $password, $timeout = 0.2 ) {
 	global $wpdb;
 
 	// dc is not used in hyperdb. This produces the desired effect of
 	// trying to connect to local servers before remote servers. Also
 	// increases time allowed for TCP responsiveness check.
-	if ( !empty($dc) && defined(DATACENTER) && $dc != DATACENTER ) {
-		if ( $read )
+	if ( ! empty( $dc ) && defined( 'DATACENTER' ) && $dc != DATACENTER ) {
```

Comment:
I just noticed that previously, this line never actually worked, as if the `DATACENTER` was `'ord'` then it'd check for a constant defined called `ord` and probably not find one.